### PR TITLE
fix: hotfix onboarding notification e2e tests

### PR DESCRIPTION
**Pull request by Prithpal-Sooriya**
## **Description**

This ensures that we do not perform any profile syncing when external services are not enabled.

Long-term, we need to discuss with the notifications team what the ideal UI we need.
1. Toggle hierarchy (we now have 4 toggles that are tied together - MetaMetrics, Profile Syncing, Notifications, and now the External Services)
2. We need to decide and handle new UI depending on this new toggle
    - Hide/show notifications panel
    - Disable or hide the other switches

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24896?quickstart=1)

## **Related issues**

Fixes:

CI - does not impact User Onboarding Journey.

NOTE - this is a hotfix, and an improved fix will be out soon (improving UI interactions, and disabling logic at the controller level)

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/pages/onboarding-flow/privacy-settings/privacy-settings.js [hunk 1]
```diff
@@ -138,8 +138,10 @@ export default function PrivacySettings() {
     setUseTransactionSimulations(isTransactionSimulationsEnabled);
     dispatch(setPetnamesEnabled(turnOnPetnames));
 
-    if (!isProfileSyncingEnabled && participateInMetaMetrics) {
-      dispatch(performSignIn());
+    if (externalServicesOnboardingToggleState) {
+      if (!isProfileSyncingEnabled && participateInMetaMetrics) {
+        dispatch(performSignIn());
```
Comment (Prithpal-Sooriya) [inline ui/pages/onboarding-flow/privacy-settings/privacy-settings.js:143]: Hotfixes to ensure we do not all these methods.

A separate PR/s will be out soon that will handle:
1. Better UI states (our toggles will either need to be disabled or not visible if the External Services toggle is not enabled)
2. Improved Logic and Controllers - ensure at a controller level that our MM services are not called when this switch is disabled.

**Review by matteoscurati — APPROVED** [review]
Thx!

File: test/e2e/helpers.js [hunk 2]
```diff
@@ -528,7 +528,7 @@ const onboardingCompleteWalletCreationWithOptOut = async (driver) => {
   await Promise.all(
     (
       await driver.findClickableElements(
-        '.toggle-button.toggle-button--on:not([data-testid="basic-functionality-toggle"] .toggle-button)',
+        '.toggle-button.toggle-button--on:not([data-testid="basic-functionality-toggle"] .toggle-button):not([data-testid="profile-sync-toggle"] .toggle-button)',
```
Comment (danjm) [inline test/e2e/helpers.js:531]: why is this change needed?

Comment (Prithpal-Sooriya) [inline test/e2e/helpers.js:531]: In the driver actions above, we have already toggled the profile-sync button off, so we don't want to turn it back on.

NOTE - we will discuss with the team to better align with the global "Basic Functionality" toggle. (just some thoughts), but if the Basic Functionality is turned off, do we want to disable or hide our notification toggles & features. May be worth discussing with other teams on how this is solved.

Comment (Prithpal-Sooriya) [reply test/e2e/helpers.js:531]: In the driver actions above (a few lines above), we have already toggled the profile-sync button off, so we don't want to turn it back on.

NOTE - we will discuss with the team to better align with the global "Basic Functionality" toggle. (just some thoughts), but if the Basic Functionality is turned off, do we want to disable or hide our notification toggles & features. May be worth discussing with other teams on how this is solved.

Comment (danjm) [reply test/e2e/helpers.js:531]: oh, I see. I guess we could also just delete the line above?

Comment (Prithpal-Sooriya) [reply test/e2e/helpers.js:531]: Since our toggle opens a modal to warn you that you are turning off, I think the lines above are valid.

The snippet above, it toggles profile sync off, and then confirms turning off profile sync.
```js
  await driver.clickElement(
    '[data-testid="profile-sync-toggle"] .toggle-button',
  );
  await driver.clickElement('[data-testid="submit-button"]');
```

Comment (Prithpal-Sooriya) [reply test/e2e/helpers.js:531]: I do get it though, this is some crazy CSS selectors here. I'll see if we can filter or do something else instead.

Comment (danjm) [reply test/e2e/helpers.js:531]: oh, sorry, I missed that part about the modal. We can leave this as is for now and clean up later

**Review by danjm — APPROVED** [review]
